cmd/atlas/internal/url/provider.go

```diff
@@ -0,0 +1,52 @@
+package url
+
+import (
+	"database/sql"
+	"errors"
+
+	"ariga.io/atlas/sql/mysql"
+	"ariga.io/atlas/sql/postgres"
+)
+
+func init() {
+	d := DefaultURLMux()
+	d.RegisterProvider("mysql", mysqlProvider)
+	d.RegisterProvider("postgres", postgresProvider)
+	d.RegisterProvider("sqlite3", sqliteProvider)
+}
+
+func mysqlProvider(dsn string) (*Atlas, error) {
+	db, err := sql.Open("mysql", dsn)
+	if err != nil {
+		return nil, err
+	}
+	drv, err := mysql.Open(db)
+	if err != nil {
+		return nil, err
+	}
+	return &Atlas{
+		DB:        db,
+		Differ:    drv.Diff(),
+		Execer:    drv.Migrate(),
+		Inspector: drv,
+	}, nil
+}
+func postgresProvider(dsn string) (*Atlas, error) {
+	db, err := sql.Open("postgres", dsn)
+	if err != nil {
+		return nil, err
+	}
+	drv, err := postgres.Open(db)
+	if err != nil {
+		return nil, err
+	}
+	return &Atlas{
+		DB:        db,
+		Differ:    drv.Diff(),
+		Execer:    drv.Migrate(),
+		Inspector: drv,
+	}, nil
+}
```

Comment:
I wonder if the code duplication can be avoided here. Maybe after we'll have a standard interface for all drivers.

Reply:
Agree, but I want to see how the other drivers will init.